barman/backup_executor.py

```diff
@@ -1634,13 +1647,10 @@ def _read_compressed_backup_label(self, backup_info):
 
         :param barman.infofile.LocalBackupInfo backup_info: backup information
         """
-        basename = os.path.join(backup_info.get_data_directory(), "base.tar")
-        with self.backup_compression.open(basename) as f:
-            self.tar = TarFile.open(fileobj=f, mode="r|")
-            for member in self.tar:
-                if member.name == "backup_label":
-                    backup_label_fileobj = self.tar.extractfile(member)
-                    return backup_label_fileobj.read().decode("utf-8")
+        basename = os.path.join(backup_info.get_data_directory(), "base")
```

Comment:
The idea of giving the archive name without the extension feels weird.  especially if we need to get the extension later in case of error (using with_suffix). it seems there is a contradiction here.
We could either discover the full name of the archive and give it to the `get_file_content`. then we do not need with_suffix method later as we would have the full archive name.
Or we could  let the `backup_compression` find the archive full name as it is actually done. Then rely on the `FileNotFoundException` to forge the new message and we would not need the actual `with_suffix` method.